app/src/main/java/com/nextcloud/ui/fileactions/FileActionsBottomSheet.kt

```diff
@@ -199,6 +199,8 @@ class FileActionsBottomSheet : BottomSheetDialogFragment(), Injectable {
     private fun displayActions(
         actions: List<FileAction>
     ) {
+        binding.fileActionsList.removeAllViews()
+
         actions.forEach { action ->
             val view = inflateActionView(action)
             binding.fileActionsList.addView(view)
```

Comment:
Works as expected.

Another approach could be to only add actions to the list when it is empty:
```suggestion
        if (binding.fileActionsList.isEmpty()) {
            actions.forEach { action ->
                val view = inflateActionView(action)
                binding.fileActionsList.addView(view)
            }
```

Reply:
I like this idea!